### PR TITLE
plot clouds in readout_characterization

### LINKED ISSUE #616: Show iq clouds in `readout_characterization` and refactor matrix
As title says

### PR DESCRIPTION
**Pull request by Edoardo-Pedicillo**
It closes #616 

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.
- [ ] Compatibility with Qibo modules (Please edit this section if the current pull request is not compatible with the following branches).
    - [ ] Qibo: `master`
    - [ ] Qibolab: `main`
    - [ ] Qibolab_platforms_qrc: `main`

### CODE REVIEW

**Comment by Jacfomg** [comment]
Have we tested that the QubitFit is doing the same as Zurich ? Because the point of this routine was to test the discrimination the machines were doing.

File: src/qibocal/protocols/characterization/readout_characterization.py [hunk 1]
```diff
@@ -171,11 +181,29 @@ def _plot(
...
     fig = go.Figure()
+    for state in range(2):
+        for measure in range(2):
+            shots = data.data[qubit, state, measure]
+
+            fig.add_trace(
+                go.Scatter(
+                    x=shots.i,
+                    y=shots.q,
+                    name=f"state {state} measure {measure}",
```
Comment (andrea-pasquale) [inline src/qibocal/protocols/characterization/readout_characterization.py:192]: Perhaps we could use a better label here.

**Review by andrea-pasquale — APPROVED** [review]
Thanks looks good to me.
Just a small comment.

**Review by Jacfomg — APPROVED** [review]
If you feel like it, I think having two plots one next to the other like the signal and phase ones would look nicer.

**Review by GabrielePalazzo — APPROVED** [review]

**Pull request merged**